packages/@react-spectrum/list/src/List.tsx

```diff
@@ -0,0 +1,139 @@
+import {ReusableView} from '@react-stately/virtualizer';
+import {useCollator, useLocale} from '@react-aria/i18n';
+import {useProvider} from '@react-spectrum/provider';
+import {Virtualizer, VirtualizerItem} from '@react-aria/virtualizer';
+
+
+export const ListContext = React.createContext(null);
+
+export function useListLayout<T>(state: ListState<T>) {
+  let {scale} = useProvider();
+  let collator = useCollator({usage: 'search', sensitivity: 'base'});
+  let layout = useMemo(() =>
+      new ListLayout<T>({
+        estimatedRowHeight: scale === 'large' ? 48 : 32,
+        estimatedHeadingHeight: scale === 'large' ? 33 : 26,
+        padding: 0, // TODO: get from DNA
+        collator
+      })
+    , [collator, scale]);
+
+  layout.collection = state.collection;
+  layout.disabledKeys = state.disabledKeys;
+  return layout;
+}
+
+function List<T extends object>(props, ref: DOMRef<HTMLDivElement>) {
+  let {
+    shouldUseVirtualFocus,
+    transitionDuration = 0
+  } = props;
+  let domRef = useDOMRef(ref);
+  let {collection} = useListState(props);
+
+  let gridCollection = new GridCollection({
+    columnCount: 1,
+    items: [...collection].map(item => ({
+      type: 'row',
+      childNodes: [{
+        ...item,
+        index: 0,
+        type: 'cell'
+      }]
+    }))
+  });
+
+
+  let state = useGridState({
+    ...props,
+    collection: gridCollection
+  });
+  let layout = useListLayout(state); // TODO should layout return nulls?
+
+  let collator = useCollator({usage: 'search', sensitivity: 'base'});
+  let {direction} = useLocale();
+  let keyboardDelegate = new GridKeyboardDelegate({
+    collection: state.collection,
+    disabledKeys: state.disabledKeys,
+    ref: domRef,
+    direction,
+    collator,
+    focusMode: 'cell'
+  });
+  let {gridProps} = useGrid({
+    keyboardDelegate,
+    ref: domRef
+  }, state);
+
+  let {styleProps} = useStyleProps(props);
+
+  // This overrides collection view's renderWrapper to support hierarchy of items in sections.
+  // The header is extracted from the children so it can receive ARIA labeling properties.
+  type View = ReusableView<Node<T>, unknown>;
+  let renderWrapper = (parent: View, reusableView: View) => (
+    <VirtualizerItem
+      key={reusableView.key}
+      reusableView={reusableView}
+      parent={parent} />
+  );
+
+  return (
+    <ListContext.Provider value={{state, keyboardDelegate}}>
+      <Virtualizer
+        {...gridProps}
+        {...styleProps}
+        ref={domRef}
+        focusedKey={state.selectionManager.focusedKey}
+        sizeToFit="height"
+        scrollDirection="vertical"
+        className={
+          classNames(
+            listStyles,
+            'react-spectrum-List',
+            styleProps.className
+          )
+        }
+        layout={layout}
+        collection={collection}
+        renderWrapper={renderWrapper}
+        transitionDuration={transitionDuration}
+        shouldUseVirtualFocus={shouldUseVirtualFocus}>
```

Comment:
Does List support virtual focus? This was added to ListBoxBase cuz ComboBox keeps "true" focus within the input field but allows users to move "virtual" focus in the listbox via the arrow keys

Reply:
unsure about this, would appreciate any input

Reply:
IMO, we can remove for now until a use case comes up where we need to add virtual focus support. Easier to add something to the api than remove (though I guess that depends if this will extend from listbox props or not)